components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -14,128 +13,176 @@ func (src *Subscription) ConvertTo(dstRaw conversion.Hub) error {
+		ExemptHandshake: src.Spec.ProtocolSettings.ExemptHandshake,
+		Qos:             src.Spec.ProtocolSettings.Qos,
+		WebhookAuth: &v1alpha2.WebhookAuth{
+			Type:         src.Spec.ProtocolSettings.WebhookAuth.Type,
+			GrantType:    src.Spec.ProtocolSettings.WebhookAuth.GrantType,
+			ClientID:     src.Spec.ProtocolSettings.WebhookAuth.ClientID,
+			ClientSecret: src.Spec.ProtocolSettings.WebhookAuth.ClientSecret,
+			TokenURL:     src.Spec.ProtocolSettings.WebhookAuth.TokenURL,
+			Scope:        src.Spec.ProtocolSettings.WebhookAuth.Scope,
+		},
+	}
+	dst.Spec.ProtocolSettings = &v1alpha2.ProtocolSettings{}
+}
+
+func (src *Subscription) setNATSSpecConfig(dst *v1alpha2.Subscription) {
 	if src.Spec.Config != nil {
-		config := src.Spec.Config
-		intVal, err := strconv.Atoi(config["maxInFlightMessages"])
-		if err == nil {
-			dst.Spec.Config = &SubscriptionConfig{}
-			dst.Spec.Config.MaxInFlightMessages = intVal
+		dst.Spec.Config = map[string]string{
+			"maxInFlightMessages": fmt.Sprint(src.Spec.Config.MaxInFlightMessages),
 		}
 	}
+}
 
-	var conditions []Condition
-	for _, condition := range src.Status.Conditions {
-		newCondition := Condition{
-			Type:               ConditionType(condition.Type),
-			Status:             condition.Status,
-			LastTransitionTime: condition.LastTransitionTime,
-			Reason:             ConditionReason(condition.Reason),
-			Message:            condition.Message,
-		}
-		conditions = append(conditions, newCondition)
+func (src *Subscription) setNATSBackendStatus(dst *v1alpha2.Subscription) {
+
+}
+func (src *Subscription) setBEBBackendStatus(dst *v1alpha2.Subscription) {
+
+	dst.Status.Backend.Ev2hash = src.Status.Ev2hash
+	dst.Status.Backend.Emshash = src.Status.Emshash
+	dst.Status.Backend.ExternalSink = src.Status.ExternalSink
+	dst.Status.Backend.FailedActivation = src.Status.FailedActivation
+	dst.Status.Backend.APIRuleName = src.Status.APIRuleName
+	dst.Status.Backend.EmsSubscriptionStatus = &v1alpha2.EmsSubscriptionStatus{
+		Status:                   src.Status.EmsSubscriptionStatus.SubscriptionStatus,
+		StatusReason:             src.Status.EmsSubscriptionStatus.SubscriptionStatusReason,
+		LastSuccessfulDelivery:   src.Status.EmsSubscriptionStatus.LastSuccessfulDelivery,
+		LastFailedDelivery:       src.Status.EmsSubscriptionStatus.LastFailedDelivery,
+		LastFailedDeliveryReason: src.Status.EmsSubscriptionStatus.LastFailedDeliveryReason,
 	}
-	dst.Status.Conditions = conditions
-	dst.Status.Ready = src.Status.Ready
-	dst.Status.CleanEventTypes = src.Status.Types
-	dst.Status.Ev2hash = src.Status.Ev2hash
-	dst.Status.Emshash = src.Status.Emshash
-	dst.Status.ExternalSink = src.Status.ExternalSink
-	dst.Status.FailedActivation = src.Status.FailedActivation
-	dst.Status.APIRuleName = src.Status.APIRuleName
+	dst.Status.Backend.EmsSubscriptionStatus = &v1alpha2.EmsSubscriptionStatus{}
+
+}
+
+func (dst *Subscription) ConvertFrom(srcRaw conversion.Hub) error {
```

Comment:
(nit) Lets keep this uncommented for now. We can see how it looks in our current version if we create a CR in the new version